src/algorithms/meta/SubDivideFunctors.h

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024, Simon Gardner
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+#pragma once
+
+#include <algorithms/geo.h>
+
+namespace eicrecon {
+
+// ----------------------------------------------------------------------------
+// Functor to split collection based on a range of values
+// ----------------------------------------------------------------------------
+template <auto MemberFunctionPtr>
+class RangeSplit {
+public:
+
+    RangeSplit(std::vector<std::pair<double,double>> ranges) : m_ranges(ranges) {};
+
+    template <typename T>
+    std::vector<int> operator()(T& instance) const {
+        std::vector<int> ids;
+        //Check if requested value is within the ranges
+        for(size_t i = 0; i < m_ranges.size(); i++){
+            if((instance.*MemberFunctionPtr)() > m_ranges[i].first && (instance.*MemberFunctionPtr)() < m_ranges[i].second){
+                ids.push_back(i);
+            }
+        }
+        return ids;
+    }
+
+private:
+    std::vector<std::pair<double,double>> m_ranges;
+
+};
+
+// ----------------------------------------------------------------------------
+// Functor to split collection based on geometry
+// ----------------------------------------------------------------------------
+class GeometrySplit {
+public:
+
+    GeometrySplit(std::vector<std::vector<long int>> ids, std::string readout, std::vector<std::string> divisions)
+    : m_ids(ids), m_readout(readout), m_divisions(divisions){};
+
+    template <typename T>
+    std::vector<int> operator()(T& instance) const {
+        if(!is_init){
```

Comment:
Use of `std::call_once` is better practice; https://en.cppreference.com/w/cpp/thread/call_once.

Reply:
This is going to need a closing parenthesis and some captured variables, but something like this:
```suggestion
        std::call_once(is_init, [](){
```

Reply:
This turned into more of a pain than it should have. Not 100% clear on the details but it seems is_init needs to be a shared_ptr potentially due to a GCC bug.